### PR TITLE
Create parent component of calendar views

### LINKED ISSUE #281: Create Parent Component of Calendar views
### **Description** 
This issue represents the work needed to build the component that will be the parent of the different calendar views available. 

Example:
Screenshot 2024-06-11 at 5 53 48 PM

### **Acceptance Criteria**
- [ ] The component should render the correct Calendar view based on the option selected by the user.

### PR DESCRIPTION
**Pull request by CDDR1**
## Description
Added the calendar parent component that will render all the views of the calendar.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] ✨ Enhancement
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #281 

## Mobile & Desktop Screenshots/Recordings

https://github.com/KyoshiNoda/ScheduleFinder/assets/62437724/d63e0ad4-279a-4da2-acaf-ce835bac0b36

### Dark Mode
Screenshot 2024-06-19 at 9 15 50 PM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## Additional information
I tried making the dropdown menu close when the user clicks somewhere else in the screen but I was getting very weird bug that wouldn't close the dropdown. I'll make a separate issue specifically for that in the interest of moving forward with the project.

### CODE REVIEW

File: client/src/components/Calendar/CalendarViewParent.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
+import { Button } from 'flowbite-react';
+import { useState } from 'react';
+import { CalendarViewEnum } from '../../enums';
+import { capitalizeWord } from '../../utils/functions';
+import { FaAngleDown, FaAngleLeft, FaAngleRight } from 'react-icons/fa6';
+
+interface Props {}
```
Comment (KyoshiNoda) [inline client/src/components/Calendar/CalendarViewParent.tsx:7]: unused interface

File: client/src/components/Calendar/CalendarViewParent.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+                <div className="absolute top-12 right-0 z-10 w-36 rounded-lg bg-white py-1 shadow dark:bg-gray-700">
+                  <ul className="text-sm text-gray-700 dark:divide-slate-600 dark:text-gray-200">
+                    <li className="py-1 md:hidden">
+                      <button className="w-full border-b border-b-gray-100 px-4 py-2 text-start text-sm text-gray-700 hover:bg-gray-100 dark:border-b-gray-600 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white">
+                        Create event
+                      </button>
+                    </li>
+                    <li className="py-1 md:hidden">
+                      <button className="w-full border-b border-b-gray-100 px-4 py-2 text-start text-sm text-gray-700 hover:bg-gray-100 dark:border-b-gray-600 dark:text-gray-200 dark:hover:bg-gray-600 dark:hover:text-white">
+                        Go to today
```
Comment (KyoshiNoda) [inline client/src/components/Calendar/CalendarViewParent.tsx:78]: Nit picky: I feel like its better to put just "Today" instead.

File: client/src/components/Calendar/CalendarViewParent.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                      </button>
+                    </li>
+                    <li>
+                      <button
+                        onClick={() => handleViewSelection(CalendarViewEnum.DAY)}
+                        className="w-full px-4 py-2 text-start hover:bg-gray-100 dark:hover:bg-gray-600 dark:hover:text-white"
+                      >
+                        Day view
```
Comment (KyoshiNoda) [inline client/src/components/Calendar/CalendarViewParent.tsx:86]: Nit Picky: I feel like its better to say "Daily view"

File: client/src/components/Calendar/CalendarViewParent.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    <svg
+      className="h-5 w-5"
+      aria-hidden="true"
+      xmlns="http://www.w3.org/2000/svg"
+      fill="currentColor"
+      viewBox="0 0 16 3"
+    >
+      <path d="M2 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Zm6.041 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3ZM14 0a1.5 1.5 0 1 1 0 3 1.5 1.5 0 0 1 0-3Z" />
+    </svg>
+  );
```
Comment (KyoshiNoda) [inline client/src/components/Calendar/CalendarViewParent.tsx:31]: Is it possible we can maybe just import the react-icons version instead? If it exists?

**Review by KyoshiNoda — CHANGES_REQUESTED** [review]
My main change: 
- I love the Calander folder! However, I believe we should have nested folders for the following:
   - `Views`: All of components for daily, weekly, monthly etc. You can add one component to represent the views for future use.
   - `Schedule`: anything schedule related
   - `TimeSlot`: anything timeslot related.

I believe as we keep building its going to get more populated.

File: client/src/components/Calendar/CalendarViewParent.tsx (hunk 3 above)
Comment (CDDR1) [reply client/src/components/Calendar/CalendarViewParent.tsx:86]: Oh yea, that was a mistake lol.

File: client/src/components/Calendar/CalendarViewParent.tsx (hunk 4 above)
Comment (CDDR1) [reply client/src/components/Calendar/CalendarViewParent.tsx:31]: I tried but didn't find one that looks as good imo. The react icon version of it is way smaller and barely visible.

**Comment by CDDR1** [comment]
> My main change:
> 
> * I love the Calander folder! However, I believe we should have nested folders for the following:
>   
>   * `Views`: All of components for daily, weekly, monthly etc. You can add one component to represent the views for future use.
>   * `Schedule`: anything schedule related
>   * `TimeSlot`: anything timeslot related.
> 
> I believe as we keep building its going to get more populated.

My thought process was that the Schedule related components are going to be replaced by Calendar ones or will become useless eventually. Let me know what you think. I'll still put them in different folders in the meantime.

File: client/src/components/Calendar/CalendarViewParent.tsx (hunk 1 above)
Comment (CDDR1) [reply client/src/components/Calendar/CalendarViewParent.tsx]: removed

**Review by KyoshiNoda — APPROVED** [review]
Approved, can you handle the merge conflict?

**Pull request merged**